diff --git a/src/pixhawk_platform.cpp b/src/pixhawk_platform.cpp
index 2d0913c..d94a29f 100644
--- a/src/pixhawk_platform.cpp
+++ b/src/pixhawk_platform.cpp
@@ -776,9 +776,9 @@ void PixhawkPlatform::px4GpsCallback(const px4_msgs::msg::SensorGps::SharedPtr m
   }
   nav_sat_fix_msg.status.service = sensor_msgs::msg::NavSatStatus::SERVICE_GPS;  // DEFAULT
 
-  nav_sat_fix_msg.latitude = msg->lat;
-  nav_sat_fix_msg.longitude = msg->lon;
-  nav_sat_fix_msg.altitude = msg->alt_ellipsoid;
+  nav_sat_fix_msg.latitude = msg->latitude_deg;
+  nav_sat_fix_msg.longitude = msg->longitude_deg;
+  nav_sat_fix_msg.altitude = msg->altitude_ellipsoid_m;
 
   if (!std::isnan(msg->eph) && !std::isnan(msg->epv)) {
     // Position uncertainty --> Diagonal known
